eng/validation/update-channel.ps1

```
@@ -15,34 +15,45 @@ function Get-Headers([string]$accept, [string]$barToken) {
-    # changes were done that could be a possibility. Also, since we already validated version $arcadeSdkVersion we move all the builds that
-    # produced it to channel $channelId
-    foreach ($asset in $assets) {
-        try {
-            $buildId = $asset[0].'buildId'
-
-            $postBuildIntoChannelApiEndpoint = "$maestroEndpoint/api/channels/$channelId/builds/$buildId/?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
-            $headers = Get-Headers 'application/json' $barToken
-        
-            Write-Host "POSTing to $postBuildIntoChannelApiEndpoint..."
-            $postResponse = Invoke-WebRequest -Uri $postBuildIntoChannelApiEndpoint -Headers $headers -Method Post
-            Write-Host "Build '$buildId' was successfully added to channel '$channelId'"
-         }
-        catch {
-            Write-Host $_
-            Write-Host $_.ScriptStackTrace
-            exit 1
-        }
-    }
-} else {
+$assets = Invoke-WebRequest -Uri $getAssetsApiEndpoint -Headers $headers | ConvertFrom-Json
+
+if (!$assets) {
     Write-Host "Asset '$arcadeSdkPackageName' with version $arcadeSdkVersion was not found"
     exit 1
-}
\ No newline at end of file
+}
+
+if ($assets.Count > 1) {
+    Write-Host "More than 1 asset matched the version '$arcadeSdkVersion' of Microsoft.DotNet.Arcade.Sdk. This is not normal. Stopping execution..."
+    exit 1
+}
+
+$getChannelsEndpoint = "$maestroEndpoint/api/channels?api-version=2018-07-16"
+$channels = Invoke-WebRequest -Uri $getChannelsEndpoint -Headers $headers | ConvertFrom-Json
+$matchedChannel = $($channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ | Select-Object -Property id)
+
+if (!$matchedChannel) {
+    Write-Host "Channel with name '$targetChannelName' was not found..."
+    exit 1    
+}
+
+$channelId = $matchedChannel.id
+
+try {
```

Why the try/catch here but not around the other calls?

---

When a build already exists in a channel we are returning 500 which for Invoke-WebRequest is considered an exception. Given the refactor we could just let the call fail since we are not in a loop anymore

---

Okay.  I have an issue I just opened re: that issue, will fix.